lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -912,29 +912,32 @@ def exec_command_output(self, command):
 
     def detect_init_system(self) -> InitMode:
         """Detect whether services are running with Docker or systemd."""
-
-        try:
-            docker_running = self.exec_command_output(
+        def _is_installed(cmd):
+            is_installed = self.exec_command(f"type {cmd} >/dev/null 2>&1") == 0
+            if not is_installed:
+                logging.info(f"{cmd} is not installed")
+            return is_installed
+
+        if _is_installed("docker"):
+            docker_magmad_running = self.exec_command_output(
                 "docker ps --filter 'name=magmad' --format '{{.Names}}'",
             ).strip() == "magmad"
-        except subprocess.CalledProcessError:
-            docker_running = False
-            logging.info("Docker is not installed")
+        else:
+            docker_magmad_running = False
 
-        try:
-            systemd_running = self.exec_command_output(
+        if _is_installed("systemctl"):
+            systemd_magmad_running = self.exec_command_output(
                 "systemctl is-active magma@magmad",
             ).strip() == "active"
-        except subprocess.CalledProcessError:
-            systemd_running = False
-            logging.info("systemd is not installed")
+        else:
+            systemd_magmad_running = False
 
-        if docker_running and systemd_running:
+        if docker_magmad_running and systemd_magmad_running:
```

Comment:
First condition is not needed at all. Actually these 3 conditions can be simplified in 2 conditions only:
```
# default to systemd if both are running, needed by feg integ tests
if systemd_magmad_running:
   return InitMode.SYSTEMD
elif docker_magmad_running:
   return InitMode.DOCKER
else:
    raise RuntimeError(
        "Magmad is not running, you have to start magmad "
        "either in Docker or systemd",
    )
```

Reply:
From a propositional logic point of view I agree - for readability I prefer the current state. Imho it especially would take a fresh reader some time to make the connection to the "default to systemd"-comment.
What do you think?

Reply:
True. I agree for the readability part. But I think here best would be to enhance the comment to elaborate a bit more. Because for condition check in code, if SYSTEMD is the default case and systemd_magmad_running is true, its actually meaningless to check for the other condition.

Reply:
Hi @VinashakAnkitAman, ok, I removed the redundant check and enhanced the comment a bit.